tensorflow/compiler/xla/service/all_reduce_key.cc

```diff
@@ -30,7 +30,15 @@ std::optional<AllReduceKey> GetAllReduceKey(const HloInstruction* instruction,
     return std::nullopt;
   }
 
-  if (instruction->to_apply()->instruction_count() != 3 ||
+  // Instruction count = 3 means the to_apply region of allreduce has this sequence:
+  // p0 = parameter(0)
+  // p1 = parameter(1)
+  // add0 = add(p0, p1)
+  // If the region has data types not supported by the inner op such as bf16 or uint8,
+  // xla will insert converts to wider precision
+  // so the instruction count of the region is 6.
+  if ((instruction->to_apply()->instruction_count() != 3 &&
+       instruction->to_apply()->instruction_count() != 6) ||
```

Comment:
With moving of the bf16 normalization pass, I am assuming these changes are not needed anymore.